src/mumble/PluginManager.cpp

```diff
@@ -139,6 +139,9 @@ void getProgramPIDs(QMultiMap<QString, unsigned long long int>& pids) {
 			pids.insert(baseName, pid);
 		}
 	}
+#else
+	g.l->log(Log::Warning, QString::fromUtf8("Retrieval of program names and PIDs only implemented for Windows and Linux"));
```

Comment:
I'd prefer a compiletime-warning but I found that there is no cross-platform way of doing so (the standard only defines `#error`). Plus this would cause the CI-builds on those systems to fail as they treat warnings as errors.

On the other hand this solution has the potential of spamming this warning at the user which is also not really what we want.

Does anyone have a better idea on how to approach this?
@davidebeatrici your thoughts on this implementation?

Reply:
I would write the missing implementations.

FreeBSD:
https://github.com/freebsd/freebsd/blob/5dab8f30b640c1560fdf77ecc73365fe2b0f22b1/bin/ps/ps.c#L491-L642

OpenBSD:
https://github.com/openbsd/src/blob/d6b8755eabe45a0eb83f5f85c8d365d23653dbc9/bin/ps/ps.c#L316-L382

macOS:
https://stackoverflow.com/a/3338118

Reply:
https://github.com/davidebeatrici/list-processes

Reply:
Implementations added :tada:

Reply:
Excellent!

I would keep the previous implementation for Linux, because it handles Wine processes and it uses C++/Qt functions.

Reply:
The new one should also handle wine processes in the same way (at least I tried to overtake the whole process) and it not using Qt is an advantage to me in this case as I'll need the names as const char* in order to pass use it...

Reply:
> The new one should also handle wine processes in the same way (at least I tried to overtake the whole process)

Sorry, I just realized there's a commit (5d18816909fc4b0597c21f8cf6fdb27a6ef780a3) doing exactly that.

> and it not using Qt is an advantage to me in this case as I'll need the names as const char* in order to pass use it..

I meant that we should use `QDir` and `QFile` instead of `opendir()`, `readdir()`, `realpath()` and `basename()`.

We can convert the string containing the name to an UTF-8 `char` array, once it's finalized.

Reply:
> Sorry, I just realized there's a commit (5d18816) doing exactly that.

Nö problem - rather complain once more often than having something missing in the code :)

> I meant that we should use QDir and QFile instead of opendir(), readdir(), realpath() and basename().

What is the advantage of this? I always thought qt has its real advantage by being cross-platform which wouldn't do any good in this case as we're explicitly writing platform dependent code :thinking:

Reply:
Cross-platform compatibility is actually not the biggest advantage: Qt provides a lot of useful APIs.

In our case the biggest advantage would be the possibility to write modern C++ code, because unfortunately filesystem functions have been introduced with C++17 (we're using C++11).

Reply:
I updated the generic BSD implementation: https://github.com/davidebeatrici/list-processes/commit/79387aff4b4ff463c376e0d91e90de8b83f85bfb

Reply:
Do we have to update the build-dependecies of Mumble to include the `util` / `kvm` library on FreeBSD / generic BSD somewhere?

Reply:
You did it the right way.

Reply:
Ah should have expressed that more cleary - I meant for documentation purposes :)

Reply:
Oh, sorry!

Both libraries are part of the operating system, they are present on a fresh installation.

Reply:
Even better :+1: